src/ProxyClient/Request/RequestQueue.php

```diff
@@ -0,0 +1,221 @@
+<?php
+
+/*
+ * This file is part of the FOSHttpCache package.
+ *
+ * (c) FriendsOfSymfony <http://friendsofsymfony.github.com/>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace FOS\HttpCache\ProxyClient\Request;
+
+use FOS\HttpCache\Exception\InvalidUrlException;
+use Http\Discovery\UriFactoryDiscovery;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\UriInterface;
+
+/**
+ * Stores each invalidation request and replicates it over all HTTP cache servers
+ *
+ * @author David de Boer <david@ddeboer.nl>
+ */
+class RequestQueue implements \Countable
```

Comment:
i was thinking RequestCollection, but it is more than a collection. RequestReplicator sounds not that wrong. or something else that explains this kind of duplicates the requests in the collection for each server, without implying that it would be responsible of actually *sending* the request. RequestDuplicator is strange because its not really duplicate, and its not distribute as each request has to go to every server...